src/nb_enum.cpp

```diff
@@ -113,6 +119,21 @@ void enum_append(PyObject *tp_, const char *name_, int64_t value_,
         fail("refusing to add duplicate key \"%s\" to enumeration \"%s\"!",
              name_, type_name(tp).c_str());
 
+    # if PY_VERSION_HEX >= 0x030B0000
+    // In Python 3.11+, update the flag and bit masks by hand,
+    // since enum._proto_member.__set_name__ is not called in this code path.
+    if (t->flags & (uint32_t) enum_flags::flag_enum) {
+        setattr(tp, "_flag_mask_", tp.attr("_flag_mask_") | val);
+
+        bool is_single_bit = (value_ != 0) && (value_ & (value_ - 1)) == 0;
+        if (is_single_bit) {
```

Comment:
the nanobind codebase skips the braces for a simple 1-liner `if` condition body.